File: frontend/src/app/models/user-create-request.ts

export class UserCreateRequest {

  public username: string;
  public password: string
  public desiredCalories: number
  public email: string

  constructor( o: any){
    this.username = o.username;
    this.password = o.password;
    this.desiredCalories = o.desiredCalories;
    this.email = o.email;
  }

  public toRequestJson() :any{
    return {
      username: this.username,
      password: this.password,
      desiredCalories: this.desiredCalories,
      email: this.email
    }
  }
}

File: backend/src/main/java/io/abnd/model/UserCreateRequest.java
package io.abnd.model;

import javax.validation.constraints.Size;
import java.util.Optional;

public class UserCreateRequest {

  @Size(max = 100)
  private String username;
  @Size(max = 100)
  private String password;
  private Optional<Integer> desiredCalories;
  @Size(max = 300)
  private String email;

  public String getUsername() {
    return username;
  }

  public void setUsername(final String username) {
    this.username = username;
  }

  public String getPassword() {
    return password;
  }

  public void setPassword(final String password) {
    this.password = password;
  }

  public Optional<Integer> getDesiredCalories() {
    return desiredCalories;
  }

  public void setDesiredCalories(final Optional<Integer> desiredCalories) {
    this.desiredCalories = desiredCalories;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(final String email) {
    this.email = email;
  }
}
